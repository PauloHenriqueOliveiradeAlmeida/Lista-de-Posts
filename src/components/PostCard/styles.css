.post {
    border: 1px solid #bbbbbb;
    border-radius: 5px;
    box-shadow: 0px 0px 10px #0000002a;
}

.post img {
    width: 100%;
}

.post-content {
    padding: 5%;
}

.post-content h1 {
    font-size: 1.5rem;
}