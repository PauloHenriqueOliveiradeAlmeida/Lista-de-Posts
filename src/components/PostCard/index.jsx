import P from "prop-types";
import "./styles.css";

function PostCard({post}) {
    return (
        <div className="post">
            <img src={post.photo} alt={post.title} />
            <div className='post-content'>
              <h2>{post.title}</h2>
              <hr />
              <p>{post.body}</p>
            </div>
          </div>
    );
}

export default PostCard;

PostCard.propTypes = {
	post: P.shape({
		photo: P.string.isRequired,
		title: P.string.isRequired,
		body: P.string.isRequired
	})
}
