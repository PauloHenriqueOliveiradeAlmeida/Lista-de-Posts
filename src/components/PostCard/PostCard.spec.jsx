import { render, screen } from "@testing-library/react";
import PostCard from ".";

const postMock = {
    photo: "image.png",
    title: "Testing PostCard component",
    body: "Mockup to test rendering of the PostCard component"
}

describe("<PostCard />", () => {
    it("should render photo, title and body in page", () => {
        render(<PostCard post={postMock}/>)

        expect(screen.getByRole("img", {name: postMock.title})).toHaveAttribute("src", postMock.photo);
        expect(screen.getByRole("heading", {name: postMock.title})).toBeInTheDocument();
        expect(screen.getByText(postMock.body)).toBeInTheDocument();
    });

    it("should match snapshot", () => {
        const { container } = render(<PostCard post={postMock} />);

        expect(container.firstChild).toMatchSnapshot();
    });
});