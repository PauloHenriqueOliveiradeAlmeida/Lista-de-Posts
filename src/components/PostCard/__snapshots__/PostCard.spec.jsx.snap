// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="Testing PostCard component"
    src="image.png"
  />
  <div
    class="post-content"
  >
    <h2>
      Testing PostCard component
    </h2>
    <hr />
    <p>
      Mockup to test rendering of the PostCard component
    </p>
  </div>
</div>
`;
